### PR TITLE
Fix document highlight column

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
fixes https://github.com/PowerShell/vscode-powershell/issues/305

We were originally computing the location by doing:

`ast.Extent.Text.IndexOf(ast.Name)`

but this is horrible because of cases like `function func { }` which break it... and not to mention it doesn't include the space before the definition... aka indentation.

Since the AST doesn't tell us where the actual name of the function starts, we need to compute it.

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919&bid=17874596)

File: src/PowerShellEditorServices/Services/Symbols/Vistors/FindReferencesVisitor.cs [hunk 1]
```diff
@@ -185,5 +181,35 @@ public override AstVisitAction VisitVariableExpression(VariableExpressionAst var
             }
             return AstVisitAction.Continue;
         }
+
+        // Computes where the start of the actual function name is.
+        private int GetStartColumnNumberFromAst(FunctionDefinitionAst ast)
```
Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Services/Symbols/Vistors/FindReferencesVisitor.cs:186]: Codacy Issue found: [Make 'GetStartColumnNumberFromAst' a static method.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919)

**Comment by TylerLeonhardt** [comment]
Ok last commit adds:

Support for silly functions like this:
image

and tests 🎉

File: test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                Assert.Equal((int)positions[positionsIndex].Line, reference.ScriptRegion.StartLineNumber);
+                Assert.Equal((int)positions[positionsIndex].Character, reference.ScriptRegion.StartColumnNumber);
+
+                positionsIndex++;
+            }
+        }
+
+        public static object[][] FindReferencesOfSymbolAtPostionData = new object[][]
+        {
+            new object[] { 2, 3, new Position[] { new Position(1, 10), new Position(2, 1) } },
```
Comment (TylerLeonhardt) [inline test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs:81]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919)

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919&bid=17874596)

File: test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+using Microsoft.PowerShell.EditorServices.Services.Symbols;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.PowerShell.EditorServices.Test.Services.Symbols
+{
+    public class AstOperationsTests
+    {
+        private readonly ITestOutputHelper _output;
```
Comment (TylerLeonhardt) [inline test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs:18]: Codacy Issue found: [Remove this unread private field '_output' or refactor the code to use its value.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919)

File: test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            foreach (SymbolReference reference in references)
+            {
+                Assert.Equal((int)positions[positionsIndex].Line, reference.ScriptRegion.StartLineNumber);
+                Assert.Equal((int)positions[positionsIndex].Character, reference.ScriptRegion.StartColumnNumber);
+
+                positionsIndex++;
+            }
+        }
+
+        public static object[][] FindReferencesOfSymbolAtPostionData = new object[][]
```
Comment (TylerLeonhardt) [inline test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs:79]: Codacy Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919)

File: test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        {
+            new object[] { 2, 3, new Position[] { new Position(1, 10), new Position(2, 1) } },
+            new object[] { 7, 18, new Position[] { new Position(4, 19), new Position(7, 3) } },
+            new object[] { 22, 13, new Position[] { new Position(12, 8), new Position(22, 5) } },
```
Comment (TylerLeonhardt) [inline test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs:83]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919)

File: test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs (hunk 4 above)
Comment (TylerLeonhardt) [inline test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs:79]: Codacy Issue found: [Change the visibility of 'FindReferencesOfSymbolAtPostionData' or make it 'const' or 'readonly'.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919)

File: test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                Assert.Equal((int)positions[positionsIndex].Character, reference.ScriptRegion.StartColumnNumber);
+
+                positionsIndex++;
+            }
+        }
+
+        public static object[][] FindReferencesOfSymbolAtPostionData = new object[][]
+        {
+            new object[] { 2, 3, new Position[] { new Position(1, 10), new Position(2, 1) } },
+            new object[] { 7, 18, new Position[] { new Position(4, 19), new Position(7, 3) } },
```
Comment (TylerLeonhardt) [inline test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs:82]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919)

File: test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
+//
```
Comment (rjmholt) [inline test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs:1]: Awesome!

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs  2
- test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919&bid=17874596)

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/PowerShellEditorServices.Test/Services/Symbols/AstOperationsTests.cs  2
- test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5451919&bid=17874596)

**Pull request merged**